File: ischia/src/main/java/cz/fb/manaus/ischia/filter/AbnormalPriceFilter.java
package cz.fb.manaus.ischia.filter;

import com.google.common.collect.Range;
import cz.fb.manaus.reactor.price.AbstractPriceFilter;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Qualifier;
import org.springframework.stereotype.Component;

@Component
public class AbnormalPriceFilter extends AbstractPriceFilter {

    @Autowired
    public AbnormalPriceFilter(@Qualifier("priceBulldoze") double priceBulldoze) {
        super(3, priceBulldoze, Range.closed(1.03d, 100d));
    }
}

File: ischia/src/main/java/cz/fb/manaus/ischia/filter/AbnormalPriceFilter.kt
package cz.fb.manaus.ischia.filter

import cz.fb.manaus.reactor.price.AbstractPriceFilter
import org.springframework.beans.factory.annotation.Qualifier
import org.springframework.context.annotation.Primary
import org.springframework.stereotype.Component

@Component
@Primary
class AbnormalPriceFilter(@Qualifier("priceBulldoze") priceBulldoze: Double) : AbstractPriceFilter(3, priceBulldoze, 1.03..100.0)
